Gems/GradientSignal/Code/Source/GradientSampler.cpp

```diff
@@ -61,45 +61,38 @@ namespace GradientSignal
 
                     ->DataElement(0, &GradientSampler::m_invertInput, "Invert Input", "")
                     ->Attribute(AZ::Edit::Attributes::ChangeNotify, &GradientSampler::ChangeNotify)
-                    ->DataElement(0, &GradientSampler::m_enableTransform, "Enable Transform", "")
-                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &GradientSampler::ChangeNotify)
+
+                    ->GroupElementToggle("Enable Transform", &GradientSampler::m_enableTransform)
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, false)
                     ->DataElement(0, &GradientSampler::m_translate, "Translate", "")
                     ->Attribute(AZ::Edit::Attributes::ReadOnly, &GradientSampler::AreTransformSettingsDisabled)
-                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &GradientSampler::ChangeNotify)
                     ->DataElement(0, &GradientSampler::m_scale, "Scale", "")
                     ->Attribute(AZ::Edit::Attributes::ReadOnly, &GradientSampler::AreTransformSettingsDisabled)
-                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &GradientSampler::ChangeNotify)
                     ->DataElement(0, &GradientSampler::m_rotate, "Rotate", "Rotation in degrees.")
                     ->Attribute(AZ::Edit::Attributes::ReadOnly, &GradientSampler::AreTransformSettingsDisabled)
-                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &GradientSampler::ChangeNotify)
 
-                    ->DataElement(0, &GradientSampler::m_enableLevels, "Enable Levels", "")
-                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &GradientSampler::ChangeNotify)
+                    ->GroupElementToggle("Enable Levels", &GradientSampler::m_enableLevels)
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, false)
                     ->DataElement(AZ::Edit::UIHandlers::Slider, &GradientSampler::m_inputMid, "Input Mid", "")
                     ->Attribute(AZ::Edit::Attributes::Min, 0.0f)
                     ->Attribute(AZ::Edit::Attributes::Max, 10.0f)
                     ->Attribute(AZ::Edit::Attributes::ReadOnly, &GradientSampler::AreLevelSettingsDisabled)
-                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &GradientSampler::ChangeNotify)
```

Comment:
No problem! In this case, the change notify function was providing a callback, and that callback is being used to refresh everything so that the logic for enable/disable would be called based on the group's boolean field. The cool part about these new toggle groups is that the property handler changes the boolean value and broadcasts the results to each of its data elements that have the ReadOnly attribute.

Reply:
I see got you, thanks for clarifying! 👍